nautobot_ssot/models.py

```diff
@@ -188,3 +190,19 @@ def get_status_class(self):
             SyncLogEntryStatusChoices.STATUS_FAILURE: "warning",
             SyncLogEntryStatusChoices.STATUS_ERROR: "danger",
         }.get(self.status)
+
+
+def _add_integrations(this_module):
+    for module in each_enabled_integration_module("models"):
+        for attr_name in dir(module):
+            attr = getattr(module, attr_name)
+            if isinstance(attr, type) and issubclass(attr, models.Model) and attr is not models.Model:
+                logger.debug("Adding model %s from %s", attr_name, module.__file__)
+                if attr_name in this_module:
+                    raise ValueError(
+                        f"Duplicate model name: {attr_name} from {module.__file__} class {attr.__class__.__name__}"
+                    )
+                this_module[attr_name] = attr
+
+
+_add_integrations(globals())
```

Comment:
Can we maybe solve this using some kind of registration pattern that allows us not to mess with `globals`? It looks like a little too much magic for my taste 😃 

If not, I would at least appreciate a docstring and some comments here.

Reply:
Other option is to use `sys.modules`, that needs to be imported. Using `globals()` seems better for me, as it doesn't need to be imported. Will add comments.